### DISCUSSION TITLE
Monthly Expense Forecasting

### DISCUSSION THREAD

**Discussion by avinashkanaujiya** [ananthakumaran/paisa#109] [Ideas]
Forecasting based on recurring transaction, it would help in knowing ahead of time, how much leeway we have at the start of the month itself.

It could be shown right next to total expense that we have currently.

(Bonus) Also maybe a table or graph that shows did we meet the forecast at month start, with the actual expense at the month end.

**Comment by ananthakumaran** [comment]
Could this be done via Budgeting? I know there is an overlap between recurring and budget, but I am not sure about adding forecasting to recurring.

**Reply by avinashkanaujiya** [reply]
By forecasting, I meant just adding up all the upcoming transaction into the total expense for the month, that would give me the  idea, about how much of my money is already committed for the month.

**Reply by avinashkanaujiya** [reply]
Doing forecasting this way, is more durable becasue it depends on hard facts (transactions that you know will happen). 

This way you don't need to add budgets separately, it could be done away with.(maybe :) )

**Reply by ananthakumaran** [reply]
Let me think about it. My main concern is that expense, recurring and budget are all interrelated and I haven't mixed them so far. Would something like `total recurring: 100$, cleared: 50$, upcoming $40, past due $10` solve your use case? there is already a legend, so adding the total won't introduce many UI changes.

**Reply by avinashkanaujiya** [reply]
Having this visible in the expense page might solve my case.
